src/SFA.DAS.Commitments.Database/Functions/CourseDatesOverLap.sql

```diff
@@ -0,0 +1,53 @@
+﻿CREATE FUNCTION [dbo].[CourseDatesOverlap](@StartDate DATETIME2, @EndDate DATETIME2, @NewStartDate DATETIME2, @NewEndDate DATETIME2)
+RETURNS SMALLINT
+AS
+BEGIN
+	DECLARE @Start INT = YEAR(@StartDate) * 100 + MONTH(@StartDate)
+	DECLARE @End INT = YEAR(@EndDate) * 100 + MONTH(@EndDate)
+	DECLARE @NewStart INT = YEAR(@NewStartDate) * 100 + MONTH(@NewStartDate)
+	DECLARE @NewEnd INT = YEAR(@NewEndDate) * 100 + MONTH(@NewEndDate)
+	DECLARE @RetVal SMALLINT
+	DECLARE @StartOverlaps BIT = 0
+	DECLARE @EndOverlaps BIT = 0
+
+	IF (@NewStart <= @Start AND @NewEnd <= @Start)
+	BEGIN
+		RETURN 0
```

Comment:
Just a suggestion, 
Instead of hard code values, can we return named variable, will make the code more readable

Reply:
OK, not sure if SQL supports CONSTs. But we could put these values if into variables if that is wanted

Reply:
either way is good

Reply:
Added some comments to make it clear what is being returned